lib/openhab/core/types/quantity_type.rb

```diff
@@ -180,50 +180,70 @@ def coerce(other)
             # def +(other)
             #   logger.trace { "#{self} + #{other} (#{other.class})" }
             #   other = other.state if other.is_a?(Core::Items::Persistence::PersistedState)
-            #   if other.is_a?(QuantityType)
+            #   case other
+            #   when QuantityType
             #     add_quantity(other)
-            #   elsif (thread_unit = DSL.unit(dimension))
-            #     if other.is_a?(DecimalType)
-            #       other = other.to_big_decimal
-            #       add_quantity(self.class.new(other, thread_unit))
-            #     elsif other.is_a?(java.math.BigDecimal)
-            #       add_quantity(self.class.new(other, thread_unit))
-            #     elsif other.respond_to?(:to_d)
-            #       other = other.to_d.to_java
-            #       add_quantity(self.class.new(other, thread_unit))
-            #     elsif other.respond_to?(:coerce) && (lhs, rhs = other.coerce(to_d))
-            #       lhs + rhs
+            #   when java.lang.Number, Numeric
```

Comment:
can you explain more of the purpose here? it seems like you're trying to avoid calling `to_d` on something that isn't a Number or a Numeric, instead opting for it to go through coerce. But that defeats the purpose of duck-typing, allowing anything that wants to be treated like a number to be treated like a number, even if it doesn't actually inherit from Number/Numeric. And AFAICT, none of the date/time types implement `to_d`, so that shouldn't be a problem

Reply:
I forgot what it was, but I think I must've overcomplicated it. Refactored to a much smaller change.

Reply:
We need to raise an exception for +/- operations against numeric outside a unit block.